src/ocean/io/select/client/FileSystemEvent.d

```diff
@@ -153,6 +208,21 @@ class FileSystemEvent : ISelectClient
 
     /***********************************************************************
 
+        Replace the handle delegate
+
+        Params:
+            handler = event handler
+
+    ***********************************************************************/
+
+    public void setHandler ( DirHandler handler )
```

Comment:
Another approach would be for the delegate to have a single argument: a `SmartUnion` of different notification types.